src/v2/keychains.js

```diff
@@ -99,6 +108,38 @@ Keychains.prototype.updatePassword = function(params, callback) {
   }).call(this).asCallback(callback);
 };
 
+/**
+ * Update the password used to decrypt a single keychain
+ * @param keychain - The keychain whose password should be updated
+ * @param oldPassword - The old password used for encrypting the key
+ * @param newPassword - The new password to be used for encrypting the key
+ * @param callback
+ * @returns {object}
+ */
+Keychains.prototype.updateSingleKeychainPassword = function({ keychain, oldPassword, newPassword }) {
+  if (!_.isString(oldPassword)) {
+    throw new Error('expected old password to be a string');
+  }
+
+  if (!_.isString(newPassword)) {
+    throw new Error('expected new password to be a string');
+  }
+
+  if (!_.isObject(keychain) || !_.isString(keychain.encryptedPrv)) {
+    throw new Error('expected keychain to be an object with an encryptedPrv property');
```

Comment:
Is it possible to throw a more specific exceptions here (ones that can be explicitly caught by the calling code in `v2/keychains.js`?

Reply:
Yes, but that is outside the scope of this change.